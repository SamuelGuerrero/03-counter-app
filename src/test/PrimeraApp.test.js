import PrimeraApp from "../PrimeraApp";
import '@testing-library/jest-dom'
import React from 'react'
import { shallow } from "enzyme";

describe('Pruebas en <PrimeraApp/>', () => { 
    
    /*test('Debe mostrar el mensaje "Hola soy Samuel"', () => { 
        const saludo = 'Hola soy Samuel';
        
        const {getByText} = render( <PrimeraApp saludo={saludo}/>)
        
        expect( getByText(saludo) ).toBeInTheDocument();
    });*/

    test('Debe de motrar <PrimeraApp/> correctamente', () => { 
        const saludo = 'Hola soy Samuel'
        const wrapper = shallow( <PrimeraApp saludo={saludo} /> ) 

        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe de mostrar el subtítulo enviado por props', () => { 
        
        const saludo = 'Hola soy Samuel';
        const subTitulo = 'Soy un subtítulo'

        const wrapper = shallow( 
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subTitulo}
            /> 
        ); 

        const textoParrafo = wrapper.find('p').text();
        console.log(textoParrafo)
        
        expect( textoParrafo ).toBe(subTitulo)

     })
});