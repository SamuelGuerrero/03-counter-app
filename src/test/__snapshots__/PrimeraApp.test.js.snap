// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe de motrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola soy Samuel
    !!! 
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
